single_iwyu.py

```diff
@@ -66,6 +66,7 @@ def perform_iwyu(fixer_path: Path, part: json, filters: List[Path], current_path
         return False
 
     old_size = linecount(preprocess_file)
+    Path.unlink(preprocess_file)
```

Comment:
This is a bit awkward. Could you turn `preprocess_file` into a `Path` object? Something like:

```
preprocess_file = Path(outfile.with_suffix('.i'))
```

(untested).

Akchually, `preprocess_file` should be a `Path` object. You can call these methods directly on `preprocess_file`.